docs/modules/sql/pages/mapping-to-kafka.adoc

```diff
@@ -0,0 +1,232 @@
+string, use the `int` and `varchar` formats:
+
+```sql
+CREATE MAPPING my_topic
+TYPE Kafka
+OPTIONS (
+    'keyFormat'='int',
+    'valueFormat'='varchar',
+    'bootstrap.servers' = '127.0.0.1:9092'
+);
+```
+
+The Kafka connector will apply a suitable serializer/deserializer automatically. For example:
+
+```sql
+'key.serializer' = 'org.apache.kafka.common.serialization.IntegerSerializer',
+'key.deserializer' = 'org.apache.kafka.common.serialization.IntegerDeserializer',
+'value.serializer' = 'org.apache.kafka.common.serialization.StringSerializer',
+'value.deserializer' = 'org.apache.kafka.common.serialization.StringDeserializer'
+```
+
+For a conversion table, see xref:data-types.adoc[].
+
+==== Avro Messages
+
+For keys or values that are in the Avro format, you need to configure different options, depending on whether you have your own Avro schema.
+
+If you don't already have an Avro schema, <<mapping-column-names,map column names>> to the fields and types that you want to use in your schema. The Kafka connector uses these column names to create its own ad-hoc schema named `jet.sql`.
+
+```sql
+CREATE MAPPING my_topic (
+    <1>
+    __key VARCHAR,
+    ticker VARCHAR,
+    amount BIGINT,
+    price DECIMAL
+)
+TYPE Kafka
+OPTIONS (
+    'keyFormat' = 'varchar',
+    'valueFormat' = 'avro',
+    'bootstrap.servers' = '127.0.0.1:9092'
+);
+```
+
+<1> Column names
+
+If you already have an Avro schema, you must:
+
+- <<mapping-column-names,Map column names>> to the fields and types of your schema.
+- Provide the URL of your schema registry in the `schema.registry.url` field.
+
+```sql
+CREATE MAPPING my_topic (
+    <1>
+    __key VARCHAR,
+    ticker VARCHAR,
+    amount BIGINT,
+    price DECIMAL
+)
+TYPE Kafka
+OPTIONS (
+    'keyFormat' = 'varchar',
+    'valueFormat' = 'avro',
+    'bootstrap.servers' = '127.0.0.1:9092',
+    'schema.registry.url' = 'http://127.0.0.1:8081/' <2>
+);
+```
+
+<1> Column names
+<2> Schema registry
+
+Your schema registry will receive entries that contain an ID for the `jet.sql` schema.
+When you write new Avro objects to the Kafka topic, the Kafka connector uses its own ad-hoc schema named `jet.sql`. The Kafka connector creates this schema from the column names that you mapped in the `CREATE MAPPING` statement.
+
+.Avro type conversion
+[cols="m,m"]
+|===
+| SQL Type | Avro Type
+
+a|`TINYINT`, `SMALLINT`, `INT`
+|INT
+
+|BIGINT
+|LONG
+
+|REAL
+|FLOAT
+
+|DOUBLE
+|DOUBLE
+
+|BOOLEAN
+|BOOLEAN
+
+a|`VARCHAR` and all other types
+|STRING
+
+|===
+
+All Avro types are a union of the `NULL` type and the actual type.
+
+==== JSON Messages
+
+If values are in the JSON format, configure the `valueFormat` field as the `json` type, and <<mapping-column-names, map column names>> to the JSON keys.
+
+```sql
+CREATE MAPPING my_topic(
+    <1>
+    __key BIGINT,
+    ticker VARCHAR,
+    amount INT)
+TYPE Kafka
+OPTIONS (
+    'keyFormat' = 'bigint',
+    'valueFormat' = 'json-flat',
```

Comment:
What is json-flat? We don't talk about it when describing serializers. Also, it doesn't match with the description above:

> If values are in the JSON format, configure the `valueFormat` field as the `json` type